### ISSUE TITLE
Automate financial report generation and verification

### ISSUE THREAD

**Issue by kargakis** [mimblewimble/grin-pm#128]
A couple of thoughts after [spending some time writting a script to verify the dev fund's income log](https://github.com/mimblewimble/grin-pm/pull/120) and also reviewing financials:

* I wouldn't fully trust public apis, there is a lot of missing or undocumented functionality spread out across different providers and I still expect we may get back false positives. I would still use public apis for verification though.
* treat the income log as the source of truth (obviously we cannot use public apis to validate Grin transactions) - eventually it would be nice in the spirit of Grin's transparent governance to run a public read-only version of the Grin wallet, once read-only wallets in Grin are a thing
* by treating the income log as the source of truth, we can start generating the quarterly financial reports based on the log, then use public apis for verification purposes

@lehnberg @ignopeverell

**Comment by lehnberg** [comment]
Yes, sounds good to me. Would you be up for making an attempt to automate the Q2 2019 report when the time comes? Even only automating the numbers parts would be nice. We could then enhance this by adding written commentary around the numbers as part of finalising the report.

**Comment by kargakis** [comment]
Agreed that's it's enough to automate just the numbers. Text should probably be manual to give more color to the data. Unfortunately, I can't commit to do this yet so the issue is up for grabs.
